sematic/api/endpoints/runs.py

```diff
@@ -415,6 +416,12 @@ def save_graph_endpoint(user: Optional[User]):
     runs = [Run.from_json_encodable(run) for run in graph["runs"]]
     for run in runs:
         logger.info("Graph update, run %s is in state %s", run.id, run.future_state)
+        if FutureState[run.future_state].is_terminal():
```

Comment:
This ensures that if the DB says the run is in a terminal state, the remote jobs for the run are actually canceled--without relying on proper handling by the resolver to ensure this.